dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04215MySQLMissingConstraints.java

```diff
@@ -0,0 +1,99 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This upgrade task will create some constraints (if they no exists) on the table workflow_task
+ * only for DB mysql, these constraints already exist on the sql file for the postgres and oracle DB
+ * and for mssql there is another task that takes care of it.
+ *
+ * @author erickgonzalez
+ * @version 4.2.0
+ * @since Aug 24, 2017
+ */
+public class Task04215MySQLMissingConstraints extends AbstractJDBCStartupTask {
+
+    @Override
+    public boolean forceRun() {
+        return true;
+    }
+
+    @Override
+    public String getPostgresScript() {
+        return null;
+    }
+
+    @Override
+    public String getMSSQLScript() {
+        return null;
+    }
+
+    @Override
+    public String getOracleScript() {
+        return null;
+    }
+
+    @Override
+    public String getMySQLScript() {
+        String sql = "";
+        final String fk_workflow_assign = "ALTER TABLE workflow_task ADD CONSTRAINT FK_workflow_assign FOREIGN KEY (assigned_to) REFERENCES cms_role (id);";
+        final String fk_workflow_task_asset = "ALTER TABLE workflow_task ADD CONSTRAINT FK_workflow_task_asset FOREIGN KEY (webasset) REFERENCES identifier (id);";
+        final String fk_workflow_step = "ALTER TABLE workflow_task ADD CONSTRAINT FK_workflow_step FOREIGN KEY (status) REFERENCES workflow_step (id);";
+        Connection conn = null;
+        final List<String> tables = new ArrayList<String>(
+                Arrays.asList("workflow_task"));
+        try {
+            conn = DbConnectionFactory.getDataSource().getConnection();
+            conn.setAutoCommit(true);
+            final List<ForeignKey> listForeignKeys = this.getForeingKeys(conn, tables, false);
+            final List<String> listForeignKeysNames = new ArrayList<String>();
+            for(ForeignKey foreignKey : listForeignKeys){
+                listForeignKeysNames.add(foreignKey.fkName());
+            }
```

Comment:
for consideration, lines 57-60 can be replaced with:
final List<String> listForeignKeysNames = listForeignKeys.stream().map(LittleObject::getName).collect(Collectors.toList());